File: HSG_THANH_HOA_2/contestants/huynh_minh_thuan/BAI4.cpp
#include <bits/stdc++.h>

using namespace std;

string solve(map<int, int> mp, int k) {
    for (auto p: mp) {
        int val = p.first;
        int c = p.second;

        if (val >= k) return "YES";

        mp[val + 1] += c / (val + 1);

        if (c % (val + 1)) return "NO";
    }

    return "YES";
}

int main() {
    ifstream inp("BAI4.inp");
    ofstream out("BAI4.out");

    int t;
    inp >> t;

    while (t--) {
        int n, k, num;
        inp >> n >> k;

        map<int, int> mp;

        while (n--) {
            inp >> num;
            if (num < k) mp[num]++;
        }

        out << solve(mp, k) << endl;
    }

    inp.close();
    out.close();

    return 0;

}

File: HSG_HAI_DUONG_22_23/contestants/bach_do/bai4.py
with open("bai4.inp","r") as fi:
    n=int(fi.readline())
    d=0
    for x in range(1,n+1):
        for y in range(x+1,n+1):
            for z in range(y+1,n+1):
                if x+y+z==n:
                    d+=1
with open("bai4.inp","w") as fo:
    fo.write(str(d))
fi.close()
fo.close()